network/p2p/middleware.go

```diff
@@ -507,16 +517,27 @@ func (m *Middleware) Subscribe(channel network.Channel) error {
 
 	topic := network.TopicFromChannel(channel, m.rootBlockID)
 
+	var authenticatePeer validatorPeerAuthenticationFunc
 	var validators []psValidator.MessageValidator
 	if !network.PublicChannels().Contains(channel) {
 		// for channels used by the staked nodes, add the topic validator to filter out messages from non-staked nodes
 		validators = append(validators,
-			// NOTE: The AuthorizedSenderValidator will assert the sender is a staked node
 			psValidator.AuthorizedSenderValidator(m.log, channel, m.ov.Identity),
 		)
+
+		// NOTE: For non-public channels the libP2P node topic validator will reject
+		// messages from unstaked nodes.
+		authenticatePeer = func(id peer.ID) bool {
+			_, ok := m.ov.Identity(id)
+			return ok
+		}
+	} else {
```

Comment:
nit: the logic "if not, else" can be confusing and require reading the code a couple times. better to flip the order so conditional is more intuitive